packages/ember-template-compiler/lib/index.js

```diff
@@ -29,6 +30,7 @@ registerPlugin('ast', TransformAngleBracketComponents);
 registerPlugin('ast', TransformInputOnToOnEvent);
 registerPlugin('ast', TransformTopLevelComponents);
 registerPlugin('ast', TransformUnescapedInlineLinkTo);
+registerPlugin('ast', AssertNoEachIn);
```

Comment:
we seem to run the tests with `_ENABLE_LEGACY_VIEW_SUPPORT = true` so I'm always registering and returning early if `_ENABLE_LEGACY_VIEW_SUPPORT` is false